src/main/java/net/snowflake/ingest/streaming/internal/StageMetadata.java

```diff
@@ -0,0 +1,117 @@
+package net.snowflake.ingest.streaming.internal;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Map;
+
+public class StageMetadata {
+  @JsonProperty(value = "locationType", required = true)
+  private String stageType; // The stage type
+
+  @JsonProperty(value = "location", required = true)
+  private String location; // The container or bucket
+
+  @JsonProperty(value = "creds", required = true)
```

Comment:
nit: I'd much rather do required field validation manually after deserialization instead of relying on the deserializer's validation via this annotation. With manual validation, the error message can be made more descriptive and have more context (like which channel was this request sent for? what was the request id?) than anything jackson can generate.

Reply:
Removed the check logic here.